### PR TITLE
fix(scylla_repository.py): make it work properly with arch

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/Hc4c6Cqy/4354-fix-ccm-to-support-new-relocatible-files-layout

From 4.6 we started releasing arm arch along with x86_64.
Because of that some of the package names had changed from `scylla-package.tar.gz` to `scylla-x86_64-package.tar.gz`
on both unstable and releases.

### CODE REVIEW

File: ccmlib/scylla_repository.py [hunk 1]
```diff
@@ -155,6 +165,7 @@ def setup(version, verbose=True):
     s3_url = ''
     type_n_version = version.split(':', 1)
     scylla_product = os.environ.get('SCYLLA_PRODUCT', 'scylla')
+    scylla_arch = os.environ.get('SCYLLA_ARCH', 'x86_64')
```
Comment (aleksbykov) [inline ccmlib/scylla_repository.py:168]: is this env variable set? Should it be passed/set to dtest docker?

File: ccmlib/scylla_repository.py [hunk 2]
```diff
@@ -117,11 +124,14 @@ def release_packages(s3_url, version):
     for package in release_packages:
         # Expected packages names (examples):
         #  'scylla-jmx-package-4.3.0-0.20210110.000585522.tar.gz'
-        #  'scylla-package-4.3.0-0.20210110.000585522.tar.gz'
+        #  'scylla-package-4.3.0-0.20210110.000585522.tar.gz' or
+        #   'scylla-x86_64-package-4.3.0-0.20210110.000585522.tar.gz'
         #  'scylla-tools-package-4.3.0-0.20210110.000585522.tar.gz'
```
Comment (aleksbykov) [inline ccmlib/scylla_repository.py:129]: lets change to 4.6.0 scylla version

File: ccmlib/scylla_repository.py (hunk 1 above)
Comment (dkropachev) [reply ccmlib/scylla_repository.py:168]: Not necessarily, currently we are targeting only x86_64, but when we would want to target arm, we should add it there.

File: ccmlib/scylla_repository.py (hunk 2 above)
Comment (dkropachev) [reply ccmlib/scylla_repository.py]: done

File: ccmlib/scylla_repository.py [hunk 3]
```diff
@@ -78,7 +78,14 @@ def release_packages(s3_url, version):
         raise RuntimeError(
             f"Failed to get release packages list for version {major_version}. URL: {s3_url}.")
 
-    all_packages = [name for name in files_in_bucket if 'jmx' in name or 'tools' in name or 'scylla-package' in name]
+    scylla_package_mark = 'scylla-package'
+
+    for file_name in files_in_bucket:
+        if f'scylla-{arch}-package' in file_name:
```
Comment (fgelcer) [inline ccmlib/scylla_repository.py:84]: this change won't allow us to run builds from 2 weeks ago or older

**Review by juliayakovlev — APPROVED** [review]
It looks ok. Did you try this with different versions?

**Comment by dkropachev** [comment]
> It looks ok. Did you try this with different versions?

Yes, on both releases and unstable relocatables

**Pull request merged**